src/main/java/com/timgroup/statsd/BlockingStatsDClient.java

```diff
@@ -0,0 +1,572 @@
+     * instantiation, and may throw an exception if that a connection cannot
+     * be established. Once a client has been instantiated in this way, all
+     * exceptions thrown during subsequent usage are consumed, guaranteeing
+     * that failures in metrics will not affect normal code execution.
+     *
+     * @param prefix
+     *     the prefix to apply to keys sent via this client
+     * @param hostname
+     *     the host name of the targeted StatsD server
+     * @param port
+     *     the port of the targeted StatsD server
+     * @param constantTags
+     *     tags to be added to all content sent
+     * @throws StatsDClientException
+     *     if the client could not be started
+     */
+    public BlockingStatsDClient(String prefix, String hostname, int port, String... constantTags) throws StatsDClientException {
+        this(prefix, hostname, port, constantTags, StatsDClientErrorHandler.NO_OP_HANDLER);
+    }
+
+    /**
+     * Create a new StatsD client communicating with a StatsD instance on the
+     * specified host and port. All messages sent via this client will have
+     * their keys prefixed with the specified string. The new client will
+     * attempt to open a connection to the StatsD server immediately upon
+     * instantiation, and may throw an exception if that a connection cannot
+     * be established. Once a client has been instantiated in this way, all
+     * exceptions thrown during subsequent usage are passed to the specified
+     * handler and then consumed, guaranteeing that failures in metrics will
+     * not affect normal code execution.
+     *
+     * @param prefix
+     *     the prefix to apply to keys sent via this client
+     * @param hostname
+     *     the host name of the targeted StatsD server
+     * @param port
+     *     the port of the targeted StatsD server
+     * @param constantTags
+     *     tags to be added to all content sent
+     * @param errorHandler
+     *     handler to use when an exception occurs during usage
+     * @throws StatsDClientException
+     *     if the client could not be started
+     */
+    public BlockingStatsDClient(String prefix, String hostname, int port, String[] constantTags, StatsDClientErrorHandler errorHandler) throws StatsDClientException {
+        if(prefix != null && prefix.length() > 0) {
+            this.prefix = String.format("%s.", prefix);
+        } else {
+            this.prefix = "";
+        }
+        this.handler = errorHandler;
+
+        /* Empty list should be null for faster comparison */
+        if(constantTags != null && constantTags.length == 0) {
+            constantTags = null;
+        }
+
+        if(constantTags != null) {
+            this.constantTagsRendered = tagString(constantTags, null);
+        } else {
+            this.constantTagsRendered = null;
+        }
+
+        try {
+            this.clientChannel = DatagramChannel.open();
+            this.address = new InetSocketAddress(hostname, port);
```

Comment:
Commented elsewhere, but this is missing the fix from https://github.com/indeedeng/java-dogstatsd-client/pull/26